### PR TITLE
Adding Kestrel connection start/stop events

### PR DESCRIPTION
**Pull request by ajay-sainy**
Adding [Connection Start ](https://github.com/dotnet/aspnetcore/blob/4b7810723c27d47cc982055c6615cccf664e2980/src/Servers/Kestrel/Core/src/Internal/Infrastructure/KestrelEventSource.cs#L73C20-L73C20) and [Connection Stop](https://github.com/dotnet/aspnetcore/blob/4b7810723c27d47cc982055c6615cccf664e2980/src/Servers/Kestrel/Core/src/Internal/Infrastructure/KestrelEventSource.cs#L91C13-L91C17) events

### CODE REVIEW

File: src/TelemetryConsumption/Kestrel/IKestrelTelemetryConsumer.cs [hunk 1]
```diff
@@ -10,6 +10,22 @@ namespace Yarp.Telemetry.Consumption;
...
 public interface IKestrelTelemetryConsumer
 {
+    /// <summary>
+    /// Called at the start of a connection.
+    /// </summary>
+    /// <param name="timestamp">Timestamp when the event was fired.</param>
+    /// <param name="connectionId">ID of the connection.</param>
+    /// <param name="localEndPoint">Local endpoint for the connection.</param>
+    /// <param name="remoteEndPoint">Remote endpoint for the connection.</param>
+    void OnConnectionStart(DateTime timestamp, string connectionId, string localEndPoint, string remoteEndPoint) { }
```
Comment (MihaZupan) [inline src/TelemetryConsumption/Kestrel/IKestrelTelemetryConsumer.cs:20]:
```suggestion
    void OnConnectionStart(DateTime timestamp, string connectionId, string? localEndPoint, string? remoteEndPoint) { }
```

File: src/TelemetryConsumption/Kestrel/KestrelEventListenerService.cs [hunk 2]
```diff
@@ -28,6 +28,30 @@ protected override void OnEvent(IKestrelTelemetryConsumer[] consumers, EventWrit
...
         {
+            case 1:
+                Debug.Assert(eventData.EventName == "ConnectionStart" && payload.Count == 3);
+                {
+                    var connectionId = (string)payload[0];
+                    var LocalEndPoint = (string)payload[1];
+                    var remoteEndPoint = (string)payload[2];
+                    foreach (var consumer in consumers)
+                    {
+                        consumer.OnConnectionStart(eventData.TimeStamp, connectionId, LocalEndPoint, remoteEndPoint);
```
Comment (MihaZupan) [inline src/TelemetryConsumption/Kestrel/KestrelEventListenerService.cs:39]:
```suggestion
                    var localEndPoint = (string?)payload[1];
                    var remoteEndPoint = (string?)payload[2];
                    foreach (var consumer in consumers)
                    {
                        consumer.OnConnectionStart(eventData.TimeStamp, connectionId, localEndPoint, remoteEndPoint);
```

**Review by MihaZupan — APPROVED** [review]
Thank you

File: src/TelemetryConsumption/Kestrel/IKestrelTelemetryConsumer.cs (hunk 1 above)
Comment (ajay-sainy) [reply src/TelemetryConsumption/Kestrel/IKestrelTelemetryConsumer.cs]: done

File: src/TelemetryConsumption/Kestrel/KestrelEventListenerService.cs (hunk 2 above)
Comment (ajay-sainy) [reply src/TelemetryConsumption/Kestrel/KestrelEventListenerService.cs]: done

**Pull request merged**